Use managed identity for backups IO in Azure Blob Storage

danipozo opened the pull request:
Also adds option to prevent ClickHouse from trying to create a non-existing container, which requires a role assignment at the storage account level.

### Changelog category (leave one):

- Improvement


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use managed identity for backups IO when using Azure Blob Storage. Add a setting to prevent ClickHouse from attempting to create a non-existent container, which requires permissions at the storage account level.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

danipozo commented:
cc @vitlibar @SmitaRKulkarni

robot-clickhouse-ci-2 commented:
<!-- automatic status comment for PR #61785 from danipozo/ClickHouse:azure-use-managed-identity-for-backups -->
*This is an automated comment for commit 1d0361b7c85d7854743c00139757aee3903963c3 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61785/1d0361b7c85d7854743c00139757aee3903963c3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

danipozo committed:
Use managed identity for backups IO in Azure Blob Storage

Also adds option to prevent ClickHouse from trying to create a non-existing container, which requires
a role assignment at the storage account level.

alexey-milovidov committed:
Update StorageAzureBlob.cpp

alexey-milovidov reviewed (approved):
The changes LGTM.
